mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3612,14 +3611,20 @@ static void generateKernel(MLIRContext *context, GenParams &genParams,
           outputLayout.getValue());
 
       SmallVector<int64_t> inDims{inputHeight, inputWidth};
-      if (nDims > 2)
+      if (nDims > 2) {
+        if (inputDepth < 1) inputDepth = 1;
```

Comment:
Simple default depth is none specified but the layout says we have that dimension.